Makefile

```diff
@@ -137,7 +137,52 @@ gofail-disable: install-gofail
 install-gofail:
 	cd tools/mod; go install go.etcd.io/gofail@${GOFAIL_VERSION}
 
-build-failpoints-release-3.5:
+# Reproduce historical issues
+
+.PHONY: reproduce-issue14370
+reproduce-issue14370: bin/etcd-v3.5.4-failpoints
+	cp bin/etcd-v3.5.4-failpoints ./bin/etcd
+	GO_TEST_FLAGS='-v --run=TestRobustness/Issue14370 --count 100 --failfast' make test-robustness
+
+.PHONY: reproduce-issue13766
+reproduce-issue13766: bin/etcd-v3.5.2-failpoints
+	cp bin/etcd-v3.5.2-failpoints ./bin/etcd
+	GO_TEST_FLAGS='-v --run=TestRobustness/Issue13766 --count 100 --failfast' make test-robustness
+
+.PHONY: reproduce-issue14685
+reproduce-issue14685: bin/etcd-v3.5.5-failpoints
+	cp bin/etcd-v3.5.5-failpoints ./bin/etcd
+	GO_TEST_FLAGS='-v --run=TestRobustness/Issue14685 --count 100 --failfast' make test-robustness
```

Comment:
I recall I raised a comment somewhere that the top level `Makefile` isn't a good place to just demonstrate the ability to the robustness test, and @ptabor suggested to move them into a script or doc inside a sub-directory.  

I still have the same comment. Let's keep the top level Makefile as simple & clean as possible.

Reply:
Don't agree with this. Robustness test targets touch files whole repo making it impossible/bothersome to put it in subdirectory. For example moving proposed targets to `tests/robustness/Makefile` would result in all paths needing to be prefixed `cd ../.. &&`.

> Let's keep the top level Makefile as simple & clean as possible. 

Don't agree, each Makefile target should be simple and clean, however it doesn't mean that root Makefile should have limited list of targets. I would argue that makefile scales pretty well with number of targets, assuming that we maintain a proper structure. For example https://github.com/kubernetes-sigs/metrics-server/blob/master/Makefile

Reply:
For me the problem is that we want to avoid adding more targets to root Makefile for it to remain `simple & clean`, however I expect that it will cause reverse effect. Make code more complicated by introducing hard to follow dependencies. Proposed targets cannot be simply moved to subdir as they depend on targets (like `GOFAIL_VERSION`, `gofail-enable`, `gofail-disable`, `test-robustness`) and directories (like `./bin`) from root directory.

Simple and clean solution here is just put those targets in root directory. Distributing makefile targets between subdirectories is the more complicated option. Makefile was just not designed for hierarchy.

Reply:
I didn't propose to add another Makefile in a sub-directory.

Reply:
Oh, I misread.

> @ptabor suggested to move them into a script or doc inside a sub-directory.

Reply:
I still would prefer to split the files. Even for sake of avoiding long content in the main directory. 
My approach would be following: 

1.  Decide how we want to run robustness tests against historical versions long-term:
2. Address in a similar way the historical cases the requires very specific etcd revision for repro. 

----

1.  I think that the build code should be as much as possible versioned together with the System-Under-Test. In this case for 3.4.x, 3.5.x, 3.6.0+ I would cherry pick to the main Makefile a rule that allows to build:  

```
make build-with-failpoints FAILPOINTS=${GO_FAIL_VERSION}
```
that as an outcome would produce `./bin/etcd-failpoints` (and potentially the other binaries)

This would allow us to have in the main file a generic rule:
```
./tmp/etcd-${}/bin/etcd-failpoints:
  (clones the repo and calls the nested build)
  
robustness-test-{...}: ./tmp/etcd-{...}/bin/etcd-failpoints 
    ETCD_BIN=./tmp/etcd-{...}/bin/etcd-failpoints  make test-robustness
```

2. For the specific repro, I would consider using the Makefile's include (https://www.gnu.org/software/make/manual/html_node/Include.html)

So having the code in `./tests/robustness/makefile.mk` and including it from the main MakeFile. 

I think that such inclusion does not causes the `cd ../..` problems still allowing to encapsulate the robustness related code with the robustness tests.